components/concept/storage/src/main/java/mozilla/components/concept/storage/HistoryStorage.kt

```diff
@@ -68,6 +68,15 @@ interface HistoryStorage : Storage {
         excludeTypes: List<VisitType> = listOf()
     ): List<VisitInfo>
 
+    /**
+     * Returns a list of the top frecent site infos limited by the given number of items
+     * sorted by most to least frecent.
+     *
+     * @param numItems the number of top frecent sites to return in the list.
```

Comment:
what happens if there are less frecent sites available than the requested number?

Reply:
In the SQL query, we do a limit based on the provided numItems. If there isn't enough entries to meet the requested number, it should still do the standard SQL operation of providing the entries it does have.